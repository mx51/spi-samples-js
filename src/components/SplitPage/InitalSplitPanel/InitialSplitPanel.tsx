import { Box, Button, Container, Grid } from '@material-ui/core';
import React, { useState } from 'react';
import Order from '../../PurchasePage/Order';
import { SummaryPanel } from '../SummaryPanel/SummaryPanel';
import { SplitOptionsPanel } from '../SplitOptionsPanel/SplitOptionsPanel';
import { SplitNumberPanel } from '../SplitNumberPanel/SplitNumberPanel';
import { SplitMode } from '../interfaces';
import { useStyles } from './InitialSplitPanel.styles';
import { SplitAmountPanel } from '../SplitAmountPanel/SplitAmountPanel';

export type InitialSplitPanelProps = {
  totalAmount: number;
  onClickNext: (splitMode: SplitMode, numberOfSplits: number, splitAmount: number) => void;
};

export const InitialSplitPanel: React.FC<InitialSplitPanelProps> = ({ totalAmount, onClickNext }) => {
  const classes = useStyles();

  const [splitMode, setSplitMode] = useState<SplitMode>('splitEvenly');
  const [numberOfSplits, setNumberOfSplits] = useState(0);
  const [splitAmount, setSplitAmount] = useState(totalAmount);

  const handleClick = () => {
    onClickNext(splitMode, numberOfSplits, splitAmount);
  };

  const isNextButtonDisabled = totalAmount <= 0 || (splitMode === 'splitEvenly' && numberOfSplits <= 0);

  return (
    <Grid container>
      <Grid item xs={4}>
        <Order disablePayNow isSubtotalEditable bottomButton="amendOrder" />
      </Grid>
      <Grid item xs={8}>
        <Container className={classes.container}>
          <SummaryPanel label="Split" amount={totalAmount} />
          <SplitOptionsPanel splitOption={splitMode} onSplitOptionChange={setSplitMode} />
          {splitMode === 'splitEvenly' && (
            <SplitNumberPanel
              totalAmount={totalAmount}
              numberOfSplits={numberOfSplits}
              onNumberOfSplitsChange={setNumberOfSplits}
            />
          )}
          {splitMode === 'splitByAmount' && (
            <SplitAmountPanel
              totalAmount={totalAmount}
              splitAmount={splitAmount}
              onSplitAmountChange={setSplitAmount}
            />
          )}
          <Box className={classes.nextButtonRow}>
            <Button
              variant="contained"
              focusRipple
              classes={{ root: classes.nextBtn, label: classes.nextBtnLabel }}
              onClick={handleClick}
              disabled={isNextButtonDisabled}
            >
              Next
            </Button>
          </Box>
        </Container>
      </Grid>
    </Grid>
  );
};
