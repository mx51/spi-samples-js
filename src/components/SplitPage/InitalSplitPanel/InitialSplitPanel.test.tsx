import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { InitialSplitPanel } from './InitialSplitPanel';
import mockWithRedux from '../../../utils/tests/common';

describe('InitialSplitPanel', () => {
  const totalAmount = 100;
  const onClickNext = jest.fn();

  it('should render the component', () => {
    const container = mockWithRedux(<InitialSplitPanel totalAmount={totalAmount} onClickNext={onClickNext} />);

    // Assert that the component is rendered
    expect(container).toMatchSnapshot();
  });

  it('should call onClickNext when the next button is clicked', () => {
    mockWithRedux(<InitialSplitPanel totalAmount={totalAmount} onClickNext={onClickNext} />);

    // Simulate a click event on the next button
    fireEvent.click(screen.getByText('Next'));

    // Assert that onClickNext is called with the correct parameters
    expect(onClickNext).toHaveBeenCalledWith('splitEvenly', 2, 100);
  });
});
