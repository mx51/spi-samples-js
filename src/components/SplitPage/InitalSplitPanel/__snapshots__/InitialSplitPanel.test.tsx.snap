// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialSplitPanel should render the component 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <div
        class="MuiPaper-root MuiBox-root MuiBox-root-22 makeStyles-root-6 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiBox-root MuiBox-root-23 makeStyles-heading-7"
        >
          <h1
            class="MuiTypography-root MuiTypography-h6"
          >
            Order
          </h1>
          <div
            class="MuiBox-root MuiBox-root-24"
          />
        </div>
        <hr
          class="MuiDivider-root"
        />
        <div
          class="MuiBox-root MuiBox-root-25 makeStyles-orderList-17"
        >
          <ul
            class="MuiList-root MuiList-padding"
          />
        </div>
        <hr
          class="MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-padding"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesEnable-26 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Subtotal
              </span>
            </div>
            <span
              aria-disabled="false"
              aria-label="Add Amount"
              class="MuiButtonBase-root MuiIconButton-root makeStyles-addAmountBtn-30"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesDisable-27 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Surcharge
              </span>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              $0.00
            </p>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesDisable-27 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Cashout
              </span>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              $0.00
            </p>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-additionalChargesDisable-27 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Tip
              </span>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              $0.00
            </p>
          </li>
          <hr
            class="MuiDivider-root makeStyles-divider-21 MuiDivider-middle"
          />
          <label
            class="MuiFormControlLabel-root makeStyles-checkbox-18"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkboxBase-20 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-34"
                  data-indeterminate="false"
                  name="Prompt for Cashout"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              <p
                class="MuiTypography-root makeStyles-checkboxLabel-19 MuiTypography-body1"
              >
                Prompt terminal for cashout
              </p>
            </span>
          </label>
          <li
            class="MuiListItem-root MuiListItem-gutters"
          >
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-total-10 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Total
              </span>
            </div>
            <p
              class="MuiTypography-root makeStyles-totalPrice-11 MuiTypography-body1"
            >
              $0.00
            </p>
          </li>
        </ul>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root makeStyles-amendBtn-13 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          href="/"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label makeStyles-actionBtnLabel-14"
          >
            &lt; Amend order
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
    >
      <div
        class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthLg"
      >
        <div
          class="MuiBox-root MuiBox-root-38 makeStyles-panel-35"
        >
          <p
            class="MuiTypography-root makeStyles-label-36 MuiTypography-body1"
          >
            Split
          </p>
          <div
            class="MuiPaper-root makeStyles-amount-37 MuiPaper-elevation1 MuiPaper-rounded"
          >
            $1.00
          </div>
        </div>
        <p
          class="MuiTypography-root makeStyles-dividerLabel-39 MuiTypography-body1"
        >
          Options
        </p>
        <hr
          class="MuiDivider-root makeStyles-divider-40"
        />
        <div
          class="MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-checked-32 Mui-checked MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-34"
                  name="splitOptions"
                  type="radio"
                  value="splitEvenly"
                />
                <div
                  class="PrivateRadioButtonIcon-root-41 PrivateRadioButtonIcon-checked-43"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-42"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Split evenly
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-34"
                  name="splitOptions"
                  type="radio"
                  value="splitByAmount"
                />
                <div
                  class="PrivateRadioButtonIcon-root-41"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-42"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Split by amount
            </span>
          </label>
        </div>
        <p
          class="MuiTypography-root makeStyles-dividerLabel-44 MuiTypography-body1"
        >
          Number of splits
        </p>
        <hr
          class="MuiDivider-root makeStyles-divider-45"
        />
        <div
          class="MuiBox-root MuiBox-root-49 makeStyles-panel-46"
        >
          <button
            aria-label="decrease number of splits"
            class="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-50 MuiIconButton-colorPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13H5v-2h14v2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <span
            class="makeStyles-numberOfSplits-47"
          >
            2
          </span>
          <button
            aria-label="decrease number of splits"
            class="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-50 MuiIconButton-colorPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <span
            class="makeStyles-amountPerSplit-48"
          >
            approximately 
            $0.50
             per split
          </span>
        </div>
        <div
          class="MuiBox-root MuiBox-root-51 makeStyles-nextButtonRow-2"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-nextBtn-3 MuiButton-contained"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label makeStyles-nextBtnLabel-4"
            >
              Next
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
